pkgs/development/libraries/dbus/default.nix

```diff
@@ -32,15 +35,22 @@ stdenv.mkDerivation rec {
     # Also applied upstream in https://gitlab.freedesktop.org/dbus/dbus/-/merge_requests/189,
     # expected in version 1.14
     ./docs-reproducible-ids.patch
-  ] ++ (lib.optional stdenv.isSunOS ./implement-getgrouplist.patch);
+  ] ++ (lib.optional stdenv.isSunOS ./implement-getgrouplist.patch)
+    # AC_PATH_XTRA doesn't seem to find X11 libs even though libX11 seems
+    # to provide valid pkg-config files. This replace AC_PATH_XTRA with
+    # PKG_CHECK_MODULES
+    ++ lib.optional stdenv.hostPlatform.isStatic
+      (fetchpatch {
+        url = "https://gitlab.freedesktop.org/dbus/dbus/-/commit/3d789d5c59abdf6ad01f62462cec5e62bbada12c.patch";
+        sha256 = "1l7nlp07p3gv9pycsml3z7w6ky5rimcz5sqmd3sr5i8mix98yaqm";
+      });
 
   postPatch = ''
-    substituteInPlace tools/Makefile.in \
-      --replace 'install-localstatelibDATA:' 'disabled:' \
+    substituteInPlace bus/Makefile.am \
+      --replace 'install-data-hook:' 'disabled:'
+    substituteInPlace tools/Makefile.am \
       --replace 'install-data-local:' 'disabled:' \
       --replace 'installcheck-local:' 'disabled:'
-    substituteInPlace bus/Makefile.in \
-      --replace '$(mkinstalldirs) $(DESTDIR)$(localstatedir)/run/dbus' ':'
```

Comment:
Why remove this?

Reply:
Because this file is overriden by autoreconfHook so I must make the substituteInPlace on the template file itself (Makefile.am) rather than on the include file (Makefile.in).

Reply:
Because the automake include files (Makefile.in) are overriden by autoreconfHook so I need to move the
patch to the template automake file directly (Makefile.am).

Reply:
Yeah, but I do not see the corresponding change in `Makefile.am`.

Reply:
Fixed.